### PR TITLE
python3Packages.openai-triton: narrow the definition of broken

### PR DESCRIPTION
**Pull request by ConnorBaker**
## Description of changes

Working on #256324, this allows us to build LLVM for the Jetson. ~~Additionally makes the AMD and NVIDIA backends mutually exclusive to reduce closure size.~~ AMD and NVIDIA backends are not mutually here. Triton actually requires that both be present: https://github.com/openai/triton/blob/768fc1fcd98ecfc0892f8982b0bb009dd7bb11ea/CMakeLists.txt#L73.

Thank you @Madouura for making this a separate derivation!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ConnorBaker** [comment]
Result of `nixpkgs-review pr 261654 --extra-nixpkgs-config '{
        allowUnfree = true;
        cudaSupport = true;
        cudaCapabilities = [ "8.9" ];
    }'` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Review by samuela — APPROVED** [review]
diff LGTM

File: pkgs/development/python-modules/openai-triton/llvm.nix [hunk 1]
```diff
@@ -67,24 +68,33 @@ stdenv.mkDerivation (finalAttrs: {
...
+      # NOTE: Unsure of how "host" would function, especially given that we might be cross-compiling.
+      llvmTargets = [ "AMDGPU" "NVPTX" ]
+        ++ lib.optionals stdenv.isAarch64 [ "AArch64" ]
+        ++ lib.optionals stdenv.isx86_64 [ "X86" ];
+    in
+    [
+      "-DLLVM_TARGETS_TO_BUILD=${lib.concatStringsSep ";" llvmTargets}"
+      "-DLLVM_ENABLE_PROJECTS=llvm;mlir"
+      "-DLLVM_INSTALL_UTILS=ON"
+    ] ++ lib.optionals (buildDocs || buildMan) [
```
Comment (SomeoneSerge) [inline pkgs/development/python-modules/openai-triton/llvm.nix:84]: Is there a way to keep the same number of spaces? 😂 I hate how bad github is at rendering this kind of diffs

Comment (ConnorBaker) [reply pkgs/development/python-modules/openai-triton/llvm.nix]: Fixed in latest -- inlined it into the string ;)

**Pull request merged**